Sources/GRPCCodeGen/CodeGenerationRequest.swift

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// Describes the services, dependencies and trivia from an IDL file,
+/// and the IDL itself through its specific serializer and deserializer.
+public struct CodeGenerationRequest {
+  /// The name of the source file containing the IDL, including the extension if applicable.
+  public var fileName: String
+
+  /// The Swift imports that the generated file depends on. The gRPC specific imports aren't required
+  /// as they will be added by default in the generated file.
+  ///
+  /// - SeeAlso: ``Dependency``.
+  public var dependencies: [Dependency]
+
+  /// Any comments at the top of the file such as documentation and copyright headers.
+  /// They will be placed at the top of the generated file.
+  public var leadingTrivia: String
+
+  /// An array of service descriptors.
+  ///
+  /// - SeeAlso: ``ServiceDescriptor``.
+  public var services: [ServiceDescriptor]
+
+  /// Closure that receives the message type and returns a string representation of the
+  /// serializer call for that message type. The result is inserted in the string representing
+  /// the generated code, where clients or servers serialize their output.
+  ///
+  /// For example: `lookupSerializer: {"ProtobufSerializer<\($0)>()"}`.
```

Comment:
Docs should start with a single sentence summary and add more detail in other paragraphs. Tools like Xcode and DocC show the first paragraph more prominently so it's quite important for it to be concise.

We can also be more specific about the requirements here: the serializer type _must_ conform to `MessageSerializer`.

```suggestion
  /// Closure that receives a message type as a `String` and returns a code snippet to 
  /// initialize a `MessageSerializer` for that type as a `String`.
  ///
  /// The result is inserted in the generated code, where clients serialize RPC inputs and
  /// servers serialize RPC outputs.
  ///
  /// For example, to serialize Protobuf messages you could specify a serializer as:
  /// ```swift
  /// request.lookupSerializer = { messageType in
  ///   "ProtobufSerializer<\(messageType)>()"
  /// }
  /// ```
```